static/js/controllers/configuration-upgrade.js

```diff
@@ -0,0 +1,110 @@
+var PouchDB = require('pouchdb-browser');
+
+// REVIEWER: are we OK hard-coding these two concepts? If not, where should
+// we put this? We don't have access to env vars in medic-webapp.
+var IS_PROD_URL = /^https:\/\/[^.]+.app.medicmobile.org\//,
+    BUILDS_DB = 'https://staging.dev.medicmobile.org/_couch/builds';
+
+angular.module('inboxControllers').controller('ConfigurationUpgradeCtrl',
+  function(
+    $log,
+    $q,
+    $scope,
+    DB
+  ) {
+
+    'use strict';
+    'ngInject';
+
+    $scope.loading = true;
+    $scope.versions = {};
+
+    DB().get('_design/medic')
+      .then(function(ddoc) {
+        $scope.deployInfo = ddoc.deploy_info;
+
+        $scope.allowBetas = $scope.allowBranches =
+          !window.location.href.match(IS_PROD_URL);
+
+        var buildsDb = new PouchDB(BUILDS_DB);
+
+        var viewLookups = [];
+
+        viewLookups.push(
+          buildsDb
+            .query('builds/branches')
+            .then(function(res) {
+              $scope.versions.branches = res.rows.sort(function(a, b) {
+                return Date.parse(b.value.build_time) -
+                       Date.parse(a.value.build_time);
+              });
+            }));
+
+        var versionRestriction;
+        var minVersion = parseVersion($scope.deployInfo && $scope.deployInfo.version);
+        if (minVersion) {
+          if (minVersion.beta) {
+            ++minVersion.beta;
+          } else {
+            ++minVersion.patch;
+          }
+
+          versionRestriction = { startkey: [ minVersion.major, minVersion.minor, minVersion.patch ] };
+        }
+
+        viewLookups.push(
+          buildsDb
+            .query('builds/betas', versionRestriction)
+            .then(function(res) {
+              $scope.versions.betas = res.rows.reverse();
+            }));
+
+        viewLookups.push(
+          buildsDb
+            .query('builds/releases', versionRestriction)
+            .then(function(res) {
+              $scope.versions.releases = res.rows.reverse();
+            }));
+
+        return $q.all(viewLookups);
+      })
+      .catch(function(err) {
+        $log.error('Error fetching available versions:', err);
+        $scope.error = 'Error fetching available versions: ' + err.message;
+      })
+      .then(function() {
+        $scope.loading = false;
+      });
+
+    $scope.upgrade = function(version) {
+      $scope.error = false;
+      $scope.upgrading = true;
+
+      window.jQuery.ajax({
```

Comment:
Don't use jQuery ever. It's why you have to have the ugly `$scope.$apply()` which is against angular best practice and causes us lots of issues. Furthermore it can't be easily mocked so it can't be easily unit tested. Use [$http.post](https://docs.angularjs.org/api/ng/service/$http#post) instead.